tests/test_metric_functions.py

```diff
@@ -0,0 +1,95 @@
+import os
+from typing import OrderedDict
+
+import numpy as np
+import yaml
+from bddl.condition_evaluation import evaluate_state
+
+import omnigibson as og
+from omnigibson.macros import gm
+from omnigibson.tasks.behavior_task import BehaviorTask
+
+# Make sure object states are enabled
+gm.ENABLE_OBJECT_STATES = True
+gm.USE_GPU_DYNAMICS = True
+
+
+def setup_env():
+
+    # Generates a BEHAVIOR Task environment using presampled cache
+    config_filename = os.path.join(og.example_config_path, "fetch_behavior.yaml")
+    cfg = yaml.load(open(config_filename, "r"), Loader=yaml.FullLoader)
+    cfg["task"]["online_object_sampling"] = False
+
+    if og.sim is not None:
+        # Make sure sim is stopped
+        og.sim.stop()
+
+    # Load the environment
+    env = og.Environment(configs=cfg)
+
+    return env
+
+
+def test_behavior_reset():
+
+    env = setup_env()
+
+    action = env.action_space.sample()
+    state, reward, terminated, truncated, info = env.step(action)
+    metrics = info["metrics"]
+
+    assert isinstance(metrics, dict)
+
+    env.reset()
+
+    # perform a step with no action
+    action = OrderedDict([("robot0", [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])])
+    state, reward, terminated, truncated, info = env.step(action)
+
+    metrics = info["metrics"]
+
+    assert metrics["steps"] == 0, "Step metric was not reset"
+    assert metrics["task_success"] == 0, "Task success metric was not reset"
+    assert metrics["wall_time"] == 0, "Wall time metric was not reset"
+    assert metrics["energy"] == 0, "Energy metric was not reset"
+    assert metrics["work"] == 0, "Work metric was not reset"
+
+
+def test_behavior_task_work_metric():
```

Comment:
can we add tests for all the metrics, e.g. wall time, energy, steps, etc.?

Reply:
env.step() 
number of time step * action_time_step